androidapp/Connexus/app/src/main/java/com/green/apt/connexus/ViewAllActivity.java

```diff
@@ -0,0 +1,133 @@
+                stream_id = jsonArr.getJSONObject(i).getLong("id");
+                stream_name = jsonArr.getJSONObject(i).getString("name");
+                streamCoverUrls.add(cover);
+                streamIds.add(stream_id);
+                streamNames.add(stream_name);
+                Log.d("ViewAllActivity","Parsed: Image #"+ i +"-- Cover: "+ cover + " -- name: " + stream_name + " -- id "+ stream_id);
+            }
+
+        }
+        catch(Exception e){
+            Log.d("ViewAllActivity", "Errored Streams: " + streams);
+            Log.e("ViewAllActivity", e.getMessage());
+        }
+
+    }
+
+    private String getAllStreams(){
+        // Instantiate the RequestQueue.
+        RequestQueue queue = Volley.newRequestQueue(this);
+        String url =getAbsoluteUrl(relativeUrl);
+
+        String response = "";
+
+        // Request a string response from the provided URL.
+        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
+                new Response.Listener<String>() {
+                    @Override
+                    public void onResponse(String response) {
+                        // Display the first 500 characters of the response string.
+                        streams = response.substring(0);
+                        parseStreams();
+                        GridView gridview = (GridView) findViewById(R.id.gridview);
+                        gridview.setAdapter(new ImageAdapter(getApplicationContext(), streamCoverUrls));
+
+                        gridview.setOnItemClickListener(new OnItemClickListener() {
+                            public void onItemClick(AdapterView<?> parent, View v,
+                                                    int position, long id) {
+                                Toast.makeText(ViewAllActivity.this, "" + streamNames.get(position),
+                                        Toast.LENGTH_SHORT).show();
+                                Intent intent = new Intent(getApplicationContext(), ViewSingleActivity.class);
+                                Bundle extras = new Bundle();
+                                extras.putString("SINGLE_STREAM_NAME", streamNames.get(position));
+                                extras.putString("SINGLE_STREAM_ID", streamIds.get(position).toString());
+                                intent.putExtras(extras);
+                                startActivity(intent);
+                            }
+                        });
+
+                    }
+                }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Log.d("ViewAllActivity","response: "+ "Response is empty or failed.");
+            }
+        });
+        // Add the request to the RequestQueue.
+        queue.add(stringRequest);
+        return response;
+    }
+
+    private static String getAbsoluteUrl(String relativeUrl) {
```

Comment:
Perhaps we could move this to BaseController and have it available to all controllers.

Reply:
this is a static function called in all my activities. We def need to move it somewhere so it exists only once and it's accessible by every activity

Reply:
As long as we create a Controller for each activity and that controller extends from BaseController I don't see why that wouldn't be possible.